pkg/kubelet/status/status_manager.go

```diff
@@ -854,6 +864,14 @@ func (m *manager) canBeDeleted(pod *v1.Pod, status v1.PodStatus) bool {
 	if pod.DeletionTimestamp == nil || kubetypes.IsMirrorPod(pod) {
 		return false
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+		// Delay deletion of Pending pods until the phase is terminal, which
+		// is expected to be set in the TerminatePod function.
+		// TODO(#116410): Transition running deleted pods to a terminal phase
+		if pod.Status.Phase == v1.PodPending {
```

Comment:
If we make the change I describe above, then this can be "If podPhase != terminalPhase" and will have nothing to do with pending, just terminal behavior.  That in the long run is where we need to be anyway.

Reply:
Simplified after merging the PRs.